examples/5_secret_storage.py

```diff
@@ -0,0 +1,412 @@
+#!/usr/bin/env python3
+
+###################################
+# WHAT IS IN THIS EXAMPLE?
+#
+# Keybase has added an encrypted key-value store; see 4_simple_storage.py for
+# more information.
+#
+# This example implements a simple bot to manage hackerspace tool rentals. It
+# shows one way you can obfuscate entryKeys (which are not encrypted) by
+# storing their HMACs, so that no one but your team (not even
+# Keybase) can know about the names of all the cool tools you have; you can do
+# something similar to hide namespaces. Additionally this example handles
+# concurrent writes using a cache to prevent one user from unintentionally
+# clobbering another user's rental updates.
+# ###################################
+
+import asyncio
+import hmac
+import json
+import logging
+import os
+import secrets
+import sys
+from base64 import b64decode, b64encode
+from enum import Enum
+from typing import Any, Dict, List, Union
+
+from pykeybasebot import Bot, EventType, KVStoreClient
+from pykeybasebot.errors import DeleteNonExistentError, RevisionError
+from pykeybasebot.types import keybase1
+
+logging.basicConfig(level=logging.DEBUG)
+
+if "win32" in sys.platform:
+    # Windows specific event-loop policy
+    asyncio.set_event_loop_policy(
+        asyncio.WindowsProactorEventLoopPolicy()  # type: ignore
+    )
+
+
+class CachedBot(Bot):
+    """
+        Custom bot maintains a cache of secrets for
+        SecretKeyKVStoreClients.
+    """
+
+    @property
+    def secret_kvstore(self):
+        if not hasattr(self, "secret"):
+            # secrets = {team: {namespace: secret}}
+            self.secrets: Dict[str, Dict[str, bytes]] = {}
+        return SecretKeyKVStoreClient(self)
+
+
+class SecretKeyKVStoreClient(KVStoreClient):
+    """
+        A KVStoreClient that hides the entryKeys from Keybase servers.
+
+        It does so by HMACing entryKeys using a per-(team, namespace) secret,
+        and storing the HMAC instead of the plaintext entryKey. This approach
+        does not handle any secret rotation, and does not expect the secret to
+        change.
+
+        The plaintext entryKey is stored in the JSON entryValue under the key
+        "_key" to enable listing; listing all keys requires getting each row
```

Comment:
"listing all keys requires getting each row..." is this necessary to state? it seems self-evident to me

Reply:
sure, removed